### PR TITLE
Features for localized documentation

### PR DESCRIPTION
**Pull request by tgamblin**
I'm in the process of creating a [new repository](/spack/localized-docs) so that we can make translations of the Spack documentation available in different languages.  Mainly looking at Japanese right now, to support Fujitsu and RIKEN teams working on Post-K, but other languages are obviously welcome.

See the [Sphinx Internationalization Docs] for details on how the process works.  I'll add a proper `README.md` on the new repo once this is all working, but the idea is to make it very easy for people to create and update translations continuously.

To support that, we need some changes to the Spack documentation build:

- [x] refactor `conf.py` so that the doc language can be injected through the `SPACK_DOC_LANGUAGE`  environment variable on readthedocs.org.

I'm also making the logic we've added to `conf.py` more "vanilla":

- [x] Add a `--header` argument to `spack commands` that prepends text to its output.
- [x] Add a `--update FILE` argument to `spack commands` so it'll only write output to `FILE` if a command has changed.
- [x] Add a similar `--update` argument to `spack list` so that it will only update the package list if it needs updating.
- [x] Remove custom code in `conf.py` and use the new command arguments instead.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Don't forget to update `share/spack/spack-completion.bash` with the new command args.

**Comment by tgamblin** [comment]
@adamjstewart: done.   also added a lot of tests to get diff coverage to 97%.

File: lib/spack/docs/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    = -E
+SPHINXOPTS    = -W
```
Comment (adamjstewart) [inline lib/spack/docs/Makefile]: What's up with this change? Don't we want the documentation tests to fail if there are any warnings?

**Comment by tgamblin** [comment]
> I don't see any modifications to share/spack/spack-completion.bash. Is that in another PR?

Oops -- push failed.  See  #11566.

File: lib/spack/docs/Makefile (hunk 1 above)
Comment (tgamblin) [reply lib/spack/docs/Makefile]: We do want this.  This is how RTD actually runs the docs so if there is an error, we'll catch it before the docs break now.

Comment (adamjstewart) [reply lib/spack/docs/Makefile]: But doesn't `-E` cause an error and `-W` only cause a warning?

Comment (tgamblin) [reply lib/spack/docs/Makefile]: No, from `sphinx-build -h`:

```
  -E                don't use a saved environment, always read all files
  -W                turn warnings into errors
```

**Pull request merged**